crates/biome_js_analyze/src/semantic_analyzers/nursery/no_global_eval.rs

```diff
@@ -0,0 +1,100 @@
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_js_syntax::{global_identifier, JsCallExpression, JsVariableDeclaration};
+use biome_rowan::{declare_node_union, AstNode};
+
+declare_rule! {
+    /// Disallow the use of `eval()`.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// eval("var a = 0");
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// f(eval);
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// var evalAlias = eval;
+    /// ```
+    ///
+    /// ### Valid
+    /// ```js
+    /// var a = 0;
+    /// ```
+    ///
+    pub(crate) NoGlobalEval {
+        version: "next",
+        name: "noGlobalEval",
+        recommended: false,
+    }
+}
+
+declare_node_union! {
+    pub(crate) JsCallExpressionOrJsVariableDeclaration = JsCallExpression | JsVariableDeclaration
+}
+
+impl Rule for NoGlobalEval {
+    type Query = Ast<JsCallExpressionOrJsVariableDeclaration>;
```

Comment:
I think that you can mimic `noGlobalIsNan` or `noFGlobalIsFinite` by querying `AnyJsExpression`
```suggestion
    type Query = Ast<AnyJsExpression>;
```

In `run` you could directly call `global_identifier` on `ctx.query()`.

Reply:
@Conaclos Thank you. By referencing the run implementation of noGlobalIsNaN, I corrected my implementation to be almost identical, which resolved the problem with the diagnostic display location and the issue of not being able to produce an error for (0, eval)('foo').